### DISCUSSION TITLE
Add additional sensors or attributes: Peak, Off-peak-1, Off-peak-2

### DISCUSSION THREAD

**Discussion by heinemannj** [mampfes/ha_epex_spot#83] [Ideas]
Add additional sensors or attributes:

- Peak
- Off-peak-1
- Off-peak-2

https://www.epexspot.com/sites/default/files/download_center_files/EPEX%20SPOT%20Indices%202019-05_final.pdf
Page 6

![image](https://github.com/mampfes/ha_epex_spot/assets/3251642/e57b97d0-7776-4ec4-a31d-d7d677379799)

Within tibber integration it is implementated in this way:

![image](https://github.com/mampfes/ha_epex_spot/assets/3251642/e92c86a5-950a-469e-a2bf-bb82728a2b9f)

Within Nordpol integration I've found the following:
https://github.com/custom-components/nordpool/blob/master/custom_components/nordpool/misc.py

```
def extract_attrs(data) -> dict:
    """extract attrs"""
    d = defaultdict(list)
    items = [i.get("value") for i in data]

    if len(data):
        data = sorted(data, key=itemgetter("start"))
        offpeak1 = [i.get("value") for i in data[0:8]]
        peak = [i.get("value") for i in data[8:20]]
        offpeak2 = [i.get("value") for i in data[20:]]

        d["Peak"] = mean(peak)
        d["Off-peak 1"] = mean(offpeak1)
        d["Off-peak 2"] = mean(offpeak2)
        d["Average"] = mean(items)
        d["Min"] = min(items)
        d["Max"] = max(items)

        return d

    return data
```

Make this sense or simply non-sense?

Note:

I've compared "EPEX SPOT", "tibber" and "Nordpol" => From my perspective **for automation approach "EPEX SPOT" is the most easiest and best integration** - THANKS

**Comment by mampfes** [comment]
Hi, it would be quite easy to add this extra information. But I have some doubts whether this information is really useful. Can you give me an example?

**Comment by mampfes** [comment]
Not to be fixed.
